src/web/pages/portlists/listpage.jsx

```diff
@@ -48,80 +70,273 @@ const ToolBarIcons = withCapabilities(
+  const previousFilter = usePreviousValue(filter);
+  const portListsSelector = useShallowEqualSelector(selector);
+  const listExportFileName = useShallowEqualSelector(state =>
+    getUserSettingsDefaults(state).getValueByName('listexportfilename'),
+  );
+  const {
+    selectionType,
+    selected: selectedEntities = [],
+    changeSelectionType,
+    select,
+    deselect,
+  } = useSelection();
+  const [sortBy, sortDir, handleSortChange] = useFilterSortBy(
+    filter,
+    changeFilter,
+  );
+  const {
+    dialogState: notificationDialogState,
+    closeDialog: closeNotificationDialog,
+    showError,
+  } = useDialogNotification();
+
+  // fetch port lists
+  const fetch = useCallback(
+    withFilter => {
+      dispatch(loadEntities(gmp)(withFilter));
+    },
+    [dispatch, gmp],
+  );
+
+  // refetch port lists with the current filter
+  const refetch = useCallback(() => {
+    fetch(filter);
+  }, [filter, fetch]);
+
+  const {entities, entitiesCounts, isLoading} = getData(
+    filter,
+    portListsSelector,
+  );
+
+  const paginationChanged = useCallback(
+    newFilter => {
+      fetch(newFilter);
+      changeFilter(newFilter);
+    },
+    [changeFilter, fetch],
+  );
+
+  const [getFirst, getLast, getNext, getPrevious] = usePagination(
+    filter,
+    entitiesCounts,
+    paginationChanged,
+  );
+  const timeoutFunc = useEntitiesReloadInterval(entities);
+  const [startReload, stopReload, hasRunningTimer] = useReload(
+    refetch,
+    timeoutFunc,
+  );
+
+  useEffect(() => {
+    // load initial data
+    if (isDefined(filter) && !isLoadingFilter) {
+      fetch(filter);
+    }
+  }, [filter, isLoadingFilter, fetch]);
+
+  useEffect(() => {
+    // reload if filter has changed
+    if (!filter.equals(previousFilter)) {
+      fetch(filter);
+    }
+  }, [filter, previousFilter, fetch]);
+
+  useEffect(() => {
+    // start reloading if tasks are available and no timer is running yet
+    if (hasValue(entities) && !hasRunningTimer) {
+      startReload();
+    }
+  }, [entities, startReload]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // stop reload on unmount
+  useEffect(() => stopReload, [stopReload]);
+
+  const closeTagsDialog = useCallback(() => {
+    renewSession();
+    setTagsDialogVisible(false);
+  }, [renewSession, setTagsDialogVisible]);
+
+  const openTagsDialog = useCallback(() => {
+    renewSession();
+    setTagsDialogVisible(true);
+  }, [renewSession, setTagsDialogVisible]);
+
+  const handleBulkDelete = useCallback(() => {
+    const entitiesCommand = gmp.portlists;
+    let promise;
+    if (selectionType === SelectionType.SELECTION_USER) {
+      promise = entitiesCommand.delete(selectedEntities);
+    } else if (selectionType === SelectionType.SELECTION_PAGE_CONTENTS) {
+      promise = entitiesCommand.deleteByFilter(filter);
+    } else {
+      promise = entitiesCommand.deleteByFilter(filter.all());
+    }
+
+    renewSession();
+
+    return promise.then(() => {
```

Comment:
Nitpick: Could we use try/catch and async/await instead of `.then`.

Reply:
Could you check twice if the code with async/await is correct now? I am used to promises ;-)